### PR TITLE
Improve architectural docs

### PR DESCRIPTION
**Pull request by Alfred-Mountfield**
## 🌟 What is the purpose of this PR?

The project has been separated into a variety of crates. This PR improves signposting to them, adds a top-level (very basic) `architecture.md` that attempts to point people towards the `cargo doc` output.

### 📜 Does this require a change to the docs?

It is one

## 🔗 Related links

- [Asana task](https://app.asana.com/0/1199548034582004/1201979309023761/f) _(internal)_

## ❓ How to test this?

1.  Checkout the branch / view the deployment
1.  Run `cargo doc --no-deps --workspace` and explore

### CODE REVIEW

**Comment by TimDiekmann** [comment]
How do we proceed with this? The Asana Task says it's blocked by Separation of Crates?

File: packages/engine/bin/hash_engine/Cargo.toml
Comment (Alfred-Mountfield) [inline packages/engine/bin/hash_engine/Cargo.toml:23]:
```suggestion
build-nng = ["experiment-control/build-nng"]

```

File: packages/engine/lib/experiment-control/src/lib.rs
Comment (Alfred-Mountfield) [inline packages/engine/lib/experiment-control/src/lib.rs:5]:
```suggestion
//! to drive the [`controller`] module, such as [`Args`] or [`Environment`]. The [`controller`] module
```

File: packages/engine/lib/simulation-control/src/controller/run.rs
Comment (Alfred-Mountfield) [inline packages/engine/lib/simulation-control/src/controller/run.rs:44]:
```suggestion
///   - Runs [Context Packages][context] in parallel
```

**Comment by nonparibus** [comment]
What's the benefit of splitting out `architecture.md` into its own file, and not just wrapping it in the `README.md`?

**Comment by Alfred-Mountfield** [comment]
> What's the benefit of splitting out architecture.md into its own file, and not just wrapping it in the README.md?

It's a very good question and one I was actually about to open on this PR for discussion :D 

`architecture.md` is a relatively common pattern that people might expect, and developers might look for. As such I think there is _some_ benefit to introducing it, but really the inclusion of it is just a remnant of when we thought it would be much bigger prior to the separation of crates. 

@TimDiekmann do you have a preference? I'm happy to throw it into the README

File: packages/engine/architecture.md
Comment (TimDiekmann) [inline packages/engine/architecture.md:7]: Let's put the `--open` at the end, then it's easier to remove it after copy/pasting or re-run a command.
```suggestion
cargo doc --workspace --no-deps --open 
```

**Review by TimDiekmann — COMMENTED** [review]
The `architecture.md` documentation is rather small and is basically saying: "generate the docs", so I'd say let's merge it into the readme.

File: packages/engine/architecture.md
Comment (Alfred-Mountfield) [reply packages/engine/architecture.md:18]: I think it's an implementation detail and we want people to go to the binary's docs first so that they then see the signposts to the relevant libs :)

**Comment by TimDiekmann** [comment]
Looks good. Will approve when CI is green

**Comment by nonparibus** [comment]
On the basis of both of your above comments, and my own views, let's move it into the `README.md` file.

**Review by TimDiekmann — DISMISSED** [review]

**Review by TimDiekmann — APPROVED** [review]

**Pull request merged**
